src/method.rs

```diff
@@ -0,0 +1,198 @@
+use std::collections::HashMap;
+
+use serde::{de, Serialize};
+use windows_core::{BSTR, HSTRING, VARIANT};
+
+use crate::{
+    de::meta::struct_name_and_fields, result_enumerator::IWbemClassWrapper,
+    ser::variant_ser::VariantStructSerializer, WMIConnection, WMIError, WMIResult,
+};
+
+impl WMIConnection {
+    fn exec_method_native_wrapper(
+        &self,
+        method_class: impl AsRef<str>,
+        object_path: impl AsRef<str>,
+        method: impl AsRef<str>,
+        in_params: HashMap<String, VARIANT>,
+    ) -> WMIResult<Option<IWbemClassWrapper>> {
+        let method_class = BSTR::from(method_class.as_ref());
+        let object_path = BSTR::from(object_path.as_ref());
+        let method = BSTR::from(method.as_ref());
+
+        unsafe {
+            let mut class_object = None;
+            self.svc.GetObject(
+                &method_class,
+                Default::default(),
+                &self.ctx.0,
+                Some(&mut class_object),
+                None,
+            )?;
+
+            match class_object {
+                Some(class) => {
+                    let mut input_signature = None;
+                    class.GetMethod(
+                        &method,
+                        Default::default(),
+                        &mut input_signature,
+                        std::ptr::null_mut(),
+                    )?;
+                    let object = match input_signature {
+                        Some(input) => {
+                            let inst = input.SpawnInstance(0)?;
+                            for (wszname, value) in in_params {
+                                inst.Put(&HSTRING::from(wszname), Default::default(), &value, 0)?;
+                            }
+                            Some(inst)
+                        }
+                        None => None,
+                    };
+
+                    let mut output = None;
+                    self.svc.ExecMethod(
+                        &object_path,
+                        &method,
+                        Default::default(),
+                        &self.ctx.0,
+                        object.as_ref(),
+                        Some(&mut output),
+                        None,
+                    )?;
+
+                    match output {
+                        Some(wbem_class_obj) => Ok(Some(IWbemClassWrapper::new(wbem_class_obj))),
+                        None => Ok(None),
+                    }
+                }
+                None => Err(WMIError::ResultEmpty),
+            }
+        }
+    }
+
+    pub fn exec_class_method<MethodClass, In, Out>(
+        &self,
+        method: impl AsRef<str>,
+        in_params: In,
+    ) -> WMIResult<Option<Out>>
```

Comment:
Done! In my implementation, I added the bound that if the function returns nothing, the only output type that will be accepted is the unit type. I think this is a reasonable compromise, but let me know if you think otherwise.